taxi/tests/test_views.py

```diff
@@ -0,0 +1,135 @@
+from django.contrib.auth import get_user_model
+from django.test import TestCase
+from django.urls import reverse
+
+from taxi.models import Manufacturer, Car, Driver
+
+MANUFACTURERS_URL = reverse("taxi:manufacturer-list")
+MANUFACTURERS_URL_WITH_SEARCH = reverse("taxi:manufacturer-list") + "?name=F"
+CARS_URL = reverse("taxi:car-list")
+CARS_URL_WITH_SEARCH = reverse("taxi:car-list") + "?model=V"
+DRIVERS_URL = reverse("taxi:driver-list")
+DRIVERS_URL_WITH_SEARCH = reverse("taxi:driver-list") + "?username=1"
+DRIVERS_URL_CREATE = reverse("taxi:driver-create")
+
+
+class PublicManufacturersTest(TestCase):
+    def test_login_required(self):
+        res = self.client.get(MANUFACTURERS_URL)
+
+        self.assertNotEqual(res.status_code, 200)
+
+
+class PublicCarsTest(TestCase):
+    def test_login_required(self):
+        res = self.client.get(CARS_URL)
+
+        self.assertNotEqual(res.status_code, 200)
+
+
+class PublicDriversTest(TestCase):
+    def test_login_required(self):
+        res = self.client.get(DRIVERS_URL)
+
+        self.assertNotEqual(res.status_code, 200)
+
+
+class PrivateManufacturerTest(TestCase):
+    def setUp(self):
+        self.user = get_user_model().objects.create_user(
+            username="testuser",
+            password="testpass"
+        )
+        self.client.force_login(self.user)
+
+    def test_retrieve_manufacturer(self):
+        response = self.client.get(MANUFACTURERS_URL)
+
+        manufacturers = Manufacturer.objects.all()
+
+        self.assertEqual(response.status_code, 200)
+
+        self.assertEqual(
+            len(response.context["manufacturer_list"]), len(manufacturers)
+        )
+
+        self.assertTemplateUsed(response, "taxi/manufacturer_list.html")
+
+    def test_search_manufacturer(self):
+        response = self.client.get(MANUFACTURERS_URL_WITH_SEARCH)
+
+        manufacturers = Manufacturer.objects.filter(name__icontains="F")
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(
+            list(response.context["manufacturer_list"]),
+            list(manufacturers)
+        )
+
+        self.assertTemplateUsed(response, "taxi/manufacturer_list.html")

```

Comment:
do u create some manufacturers for this tests? all method will also return blank list here)

Reply:
fix everywhere

Reply:
Sorry for being late! It is fixed now.